src/EditorFeatures/CSharp/NavigationBar/CSharpNavigationBarItemService.cs

```diff
@@ -53,44 +35,47 @@ public CSharpNavigationBarItemService()
 
         public override async Task<IList<NavigationBarItem>> GetItemsAsync(Document document, CancellationToken cancellationToken)
         {
-            var typesInFile = await GetTypesInFileAsync(document, cancellationToken).ConfigureAwait(false);
+            var documentSymbolsSerivce = document.GetRequiredLanguageService<IDocumentSymbolsService>();
+            var membersInDocument = await documentSymbolsSerivce.GetSymbolsInDocumentAsync(document, DocumentSymbolsOptions.TypesAndMethodsOnly, cancellationToken).ConfigureAwait(false);
+            if (membersInDocument.IsEmpty || cancellationToken.IsCancellationRequested)
+            {
+                return SpecializedCollections.EmptyList<NavigationBarItem>();
+            }
+
             var tree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-            return GetMembersInTypes(tree, typesInFile, cancellationToken);
+            // If we got members from the symbol service, there had to have been a tree
+            Debug.Assert(tree is not null);
```

Comment:
```suggestion
```

This is exported for C#.  it's an invariant that C#/Vb always have syntax trees for all docs.

Reply:
these comments both still apply.